sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -97,15 +97,20 @@ abstract class Expression extends TreeNode[Expression] {
     ctx.subExprEliminationExprs.get(this).map { subExprState =>
       // This expression is repeated which means that the code to evaluate it has already been added
       // as a function before. In that case, we just re-use it.
-      val code = s"/* ${toCommentSafeString(this.toString)} */"
-      ExprCode(code, subExprState.isNull, subExprState.value)
+      val placeHolder = s"/*{${ctx.freshName("comment_placeholder")}}*/"
```

Comment:
Actually, `}` is really needed.
If we have place holders like `comment_placeholder1` and `comment_placeholder12`, and the actual comment `Hello` corresponds to `comment_placeholder1`, `comment_placeholder1` will be replaced with `Hello` but also `comment_placeholder12` will be replaced with `Hello2`.